tango/step_caches/local_step_cache.py

```diff
@@ -117,17 +149,27 @@ def __setitem__(self, step: Step, value: Any) -> None:
             raise
 
     def __len__(self) -> int:
-        return sum(1 for _ in self.dir.glob("*/cache-metadata.json"))
+        return sum(1 for _ in self.dir.glob(f"*/{self.METADATA_FILE_NAME}"))
 
-    def step_dir(self, step_or_unique_id: Union[Step, str]) -> Path:
+    def step_dir(self, step_or_unique_id: Union[Step, StepInfo, str]) -> Path:
         """Returns the directory that contains the results of the step.
 
         You can use this even for a step that's not cached yet. In that case it will return the directory where
         the results will be written."""
-        if isinstance(step_or_unique_id, Step):
-            if not step_or_unique_id.cache_results:
+        if isinstance(step_or_unique_id, (Step, StepInfo)):
+            cacheable = (
+                step_or_unique_id.cache_results
+                if isinstance(step_or_unique_id, Step)
+                else step_or_unique_id.cacheable
+            )
+            if not cacheable:
+                class_name = (
+                    step_or_unique_id.__class__.__name__
+                    if isinstance(step_or_unique_id, Step)
+                    else step_or_unique_id.step_class_name
+                )
                 raise RuntimeError(
-                    f"Uncacheable steps (like '{step_or_unique_id.name}') don't have step directories."
+                    f"Uncacheable steps (like '{class_name}') don't have step directories."
```

Comment:
I'm OK with this, but in principle, one instance of a step class could be cacheable, and another could be not cacheable.